src/main/scala/intellij/haskell/editor/HaskellCompletionContributor.scala

```diff
@@ -182,6 +184,10 @@ class HaskellCompletionContributor extends CompletionContributor {
                 resultSet.addAllElements(findTopLevelTypeSignatureLookupElements(psiFile).asJavaCollection)
               }
 
+              if (HaskellConsoleView.isConsole(psiFile)) {
```

Comment:
> Ok, maybe another completionContributor for Haskell language, is that possible to have multiple completionContributor for a single language?

It's not necessary to have another completion contributor. From the entrance you have go to directly to another flow in which you probably can reuse already existing code and also the already cached exports. You will to set up in the same way, except that in the repl the loaded file is the context. Also depending with which target you start the repl.